### PR TITLE
Add studio.eufonia.EufoniaClient

### PR DESCRIPTION
**Pull request by KernelFreeze**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly. Eufonia Client is a custom Minecraft Launcher for modded Minecraft that allows the user to join events made by Eufonia Studio, for example SquidCraft Games 3, Dedsafio Minecraft, etc. The Launcher uses the standard Microsoft Authentication scheme and never sends the Microsoft Account Password or login details to our servers.
- [X] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: studio.eufonia.EufoniaClient.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        path: patches/0003-Don-t-crash-on-calls-to-focus-or-icon.patch
+      - type: patch
+        path: patches/0005-Add-warning-about-being-an-unofficial-patch.patch
+      - type: patch
+        path: patches/0007-Platform-Prefer-Wayland-over-X11.patch
+    cleanup:
+      - /include
+      - /lib/cmake
+      - /lib/pkgconfig
+  - name: binary
```
Comment (hfiguiere) [inline studio.eufonia.EufoniaClient.yml:61]:
```suggestion
  - name: eufonia
```

File: studio.eufonia.EufoniaClient.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    build-commands:
+      - ar -x Eufonia-Client-linux-v2.0.1.deb
+      - tar -xf data.tar.gz
+      - 'install -Dm755 usr/bin/eufonia-client /app/bin/eufonia-client'
+      - 'install -Dm755 usr/bin/cloudflared /app/bin/cloudflared'
+      - install -Dm644 usr/share/applications/Eufonia\ Client.desktop /app/share/applications/studio.eufonia.EufoniaClient.desktop
+      - install -Dm644 usr/share/icons/hicolor/128x128/apps/eufonia-client.png /app/share/icons/hicolor/128x128/apps/studio.eufonia.EufoniaClient.png
+      - install -Dm644 usr/share/icons/hicolor/256x256@2/apps/eufonia-client.png /app/share/icons/hicolor/256x256@2/apps/studio.eufonia.EufoniaClient.png
+      - install -Dm644 usr/share/icons/hicolor/32x32/apps/eufonia-client.png /app/share/icons/hicolor/32x32/apps/studio.eufonia.EufoniaClient.png
+      - sed -i 's/Icon=eufonia-client/Icon=studio.eufonia.EufoniaClient/g' /app/share/applications/studio.eufonia.EufoniaClient.desktop
```
Comment (hfiguiere) [inline studio.eufonia.EufoniaClient.yml:80]: don't use `sed`. Use `desktop-file-edit`

File: studio.eufonia.EufoniaClient.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      - -DBUILD_SHARED_LIBS:BOOL=ON
+      - -DGLFW_USE_WAYLAND:BOOL=ON
+      - -DGLFW_BUILD_DOCS:BOOL=OFF
+    sources:
+      - type: git
+        url: https://github.com/glfw/glfw.git
+        commit: 3fa2360720eeba1964df3c0ecf4b5df8648a8e52
+      - type: patch
+        path: patches/0003-Don-t-crash-on-calls-to-focus-or-icon.patch
```
Comment (hfiguiere) [inline studio.eufonia.EufoniaClient.yml:52]: these patches are missing.

File: studio.eufonia.EufoniaClient.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    cleanup:
+      - /include
+      - /lib/cmake
+      - /lib/pkgconfig
+  - name: binary
+    buildsystem: simple
+    sources:
+      - type: file
+        url: https://client.eufonia.studio/api/v2/download?file_type=installer&platform=debian-x86_64&version=2.0.1
+        dest-filename: Eufonia-Client-linux-v2.0.1.deb
```
Comment (bbhtt) [inline studio.eufonia.EufoniaClient.yml:66]:
```suggestion
        dest-filename: Eufonia-Client.deb
```

File: studio.eufonia.EufoniaClient.yml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        sha256: 69561e9b820492c4e0e76410c3e463ea0b365359cafce1c105c5bc50113f1fe2
+        only-arches: [x86_64] #This source is only used on x86_64 Computers
+      - type: file
+        path: studio.eufonia.EufoniaClient.metainfo.xml
+    build-commands:
+      - ar -x Eufonia-Client-linux-v2.0.1.deb
```
Comment (bbhtt) [inline studio.eufonia.EufoniaClient.yml:72]:
```suggestion
      - ar -x Eufonia-Client.deb
```

File: studio.eufonia.EufoniaClient.metainfo.xml
Comment (bbhtt) [inline studio.eufonia.EufoniaClient.metainfo.xml:1]: if you are the upstream, this file should be inside the deb.

File: studio.eufonia.EufoniaClient.yml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+      - type: patch
+        path: patches/0003-Don-t-crash-on-calls-to-focus-or-icon.patch
+      - type: patch
+        path: patches/0005-Add-warning-about-being-an-unofficial-patch.patch
+      - type: patch
+        path: patches/0007-Platform-Prefer-Wayland-over-X11.patch
+    cleanup:
+      - /include
+      - /lib/cmake
+      - /lib/pkgconfig
```
[the remark this answers is not in the document]
Comment (KernelFreeze) [reply studio.eufonia.EufoniaClient.yml:60]: What do you mean by toplevel? Should I invoke the cleanup at the end or do you mean something else?

[the remark this answers is not in the document]
Comment (bbhtt) [reply studio.eufonia.EufoniaClient.yml:60]: add it after finish-args, it is at the module level right now affecting this particular module.

File: studio.eufonia.EufoniaClient.metainfo.xml
Comment (KernelFreeze) [reply studio.eufonia.EufoniaClient.metainfo.xml:1]: I added it to the 2.0.3 version of the .deb package

**Comment by KernelFreeze** [comment]
I did all the requested changes, is there anything else that you want me to change?

File: studio.eufonia.EufoniaClient.yml [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+runtime: org.gnome.Platform
+runtime-version: '47'
+sdk: org.gnome.Sdk
+tags:
+  - proprietary
+
+command: eufonia-client
+finish-args:
+  - --device=dri # Hardware acceleration
```
Comment (bbhtt) [inline studio.eufonia.EufoniaClient.yml:11]:
```suggestion
```

you have all already.

Comment (KernelFreeze) [reply studio.eufonia.EufoniaClient.yml:11]: I resolved this by removing --device=all

File: studio.eufonia.EufoniaClient.yml [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+  - --device=input # Game controllers
```
Comment (hfiguiere) [inline studio.eufonia.EufoniaClient.yml:12]: this won't work. It's not available anywhere.

Comment (bbhtt) [reply studio.eufonia.EufoniaClient.yml:12]: yes for now

Comment (KernelFreeze) [reply studio.eufonia.EufoniaClient.yml:12]: Okay! Thank you for the information

**Comment by bbhtt** [comment]
bot, build studio.eufonia.EufoniaClient

**Comment by KernelFreeze** [comment]
I saw that the CI job failed so I updated the metainfo path, I think that it's fixed now.

**Comment by bbhtt** [comment]
bot, build studio.eufonia.EufoniaClient

**Comment by bbhtt** [comment]
can you show a screenrecord of the app after logging in?

**Comment by KernelFreeze** [comment]
Sure, here is it
image

**Pull request closed without merging**